relayer/message_relayer.go

```diff
@@ -64,22 +64,19 @@ type messageRelayer struct {
 }
 
 func newMessageRelayer(
-	logger logging.Logger,
-	metrics *MessageRelayerMetrics,
 	relayer *Relayer,
 	warpMessage *warp.UnsignedMessage,
 	destinationBlockchainID ids.ID,
 	messageResponseChan chan message.InboundMessage,
-	messageCreator message.Creator,
 ) *messageRelayer {
 	return &messageRelayer{
 		relayer:                 relayer,
 		warpMessage:             warpMessage,
 		destinationBlockchainID: destinationBlockchainID,
 		messageResponseChan:     messageResponseChan,
-		logger:                  logger,
-		metrics:                 metrics,
-		messageCreator:          messageCreator,
+		logger:                  relayer.logger,
+		metrics:                 relayer.metrics,
+		messageCreator:          relayer.messageCreator,
```

Comment:
We can simplify the `messageRelayer` struct by removing these fields altogether and referencing them via the `relayer` field.